PREREQUISITE.md

```diff
@@ -183,76 +183,74 @@ The sample code below is an example of adding permission through the UI. Please
 Let's look up the code.
 
 ```java
-package com.example.euphonytest;
+package com.exmaple.euphonytest;
 
-import androidx.annotation.NonNull;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.core.content.ContextCompat;
 
 import android.Manifest;
-import android.content.pm.PackageManager;
 import android.os.Build;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
 
 public class MainActivity extends AppCompatActivity {
+    private ActivityResultLauncher requestPermissionLauncher;
     private Button button;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        requestPermissionLauncher = registerForActivityResult(
+                new ActivityResultContracts.RequestPermission(),
+                new ActivityResultCallback<Boolean>() {
+                    @Override
+                    public void onActivityResult(Boolean result) {
+                        if (result) {
+                            // PERMISSION GRANTED
+                        } else {
+                            // PERMISSION DENIED
+                        }
+                    }
+                });
+
         button = findViewById(R.id.button2);
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                int permission = ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO);
-
-                if (permission == PackageManager.PERMISSION_DENIED) {
-                    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                        requestPermissions(new String[]{
-                                Manifest.permission.RECORD_AUDIO}, 1000);
-                    }
+                if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    final String permission = Manifest.permission.RECORD_AUDIO;
+                    requestPermissionLauncher.launch(permission);
```

Comment:
I use ActivityResultLauncher instead of requestPermissions.
I'm not sure this is best way, so please tell me if there are other good ways.